pkg/volume/volume.go

```diff
@@ -132,11 +132,39 @@ type Attacher interface {
 	Attach() error
 }
 
+type PhysicalAttacher interface {
+	// WaitForAttach examines the file system for the device; if
+	// found that means the disk is attached and this function
+	// returns otherwise it polls until the device appears. If the
+	// device is not attached within the timeout period an error
+	// is returned.
+	WaitForAttach() error
```

Comment:
> If I make those changes I don't think we'll need MountDevice() anymore the kubelet can just mount the device to the global path using its own mounter.

Okay changing my mind about this; the global mount has minor variations between volume plugins. Some inspect the volume spec to check for read only volumes, and RBD implements fencing etc.